lib/layout/image_placeholder.dart

```diff
@@ -54,23 +54,48 @@ class FadeInImagePlaceholder extends StatelessWidget {
   /// See [Image.fit].
   final BoxFit fit;
 
+  @override
+  _FadeInImagePlaceholderState createState() => _FadeInImagePlaceholderState();
+}
+
+class _FadeInImagePlaceholderState extends State<FadeInImagePlaceholder> {
+  bool _isLoaded = false;
+
+  @override
+  void initState() {
+    super.initState();
+    // Once the image is loaded, we no longer need to show a placeholder
+    widget.image
+        .resolve(ImageConfiguration.empty)
```

Comment:
This will wait for the asset with the given configuration, which is not necessarily the one that ends up being shown on screen?